### DISCUSSION TITLE
__setattr__ not called on Metaclass

### DISCUSSION THREAD

**Discussion by zwadar** [IronLanguages/ironpython3#1660] [Q&A] [answered]
Hi,
I'm using 3.4.0 and trying to use __setattr__ on a metaclass but it doesn't seem to be called. As far as I can see this should be supported and I don't see any issues reported, so I was wondering if I might be doing something wrong?

    class Test(type):
        def __new__(cls, name, bases, namespace):
            cls = super(Test, cls).__new__(cls, name, bases, namespace)
            return cls        
   
        def __setattr__(cls, attr, value):
            print("Called Set Attr")
            super(Test, self).__setattr__(attr, value)
    
    class TestMeta(object, metaclass=Test):
        aaa = 10
    
    TestMeta.aaa = 80
    print(TestMeta.aaa)

but the printed output is just

   80

Can I somehow overcome it? This is pretty crucial to me.

Thank you

**Comment by slozier** [comment, marked as answer]
Thanks, this one sounds like a bug so I've filed an issue: https://github.com/IronLanguages/ironpython3/issues/1661

Unfortunately I can't think of a workaround for this...

**Comment by zwadar** [comment]
Thanks, I had changed my code to use instances rather then class directly for time being.
